src/main/java/org/dependencytrack/persistence/QueryManager.java

```diff
@@ -880,8 +880,8 @@ public PaginatedResult getComponents(final Project project, final boolean includ
         return getComponentQueryManager().getComponents(project, includeMetrics);
     }
 
-    public PaginatedResult getComponents(final Project project, final boolean includeMetrics, final boolean onlyOutdated) {
-        return getComponentQueryManager().getComponents(project, includeMetrics, onlyOutdated);
+    public PaginatedResult getComponents(final Project project, final boolean includeMetrics, final boolean onlyOutdated, final boolean onlyDirect) {
+        return getComponentQueryManager().getComponents(project, includeMetrics, onlyOutdated, onlyDirect);
```

Comment:
@walterdeboer Now seeing this PR.  If i understand this right, this is enabling a new toggle switch to show only direct dependencies, hence why the previous PR #2537 has code comments about direct dependencies only?

Reply:
Hmm, I think i've mixed up some changes from #2568...

Reply:
I've mixed up the code in #2537 I hope i've got it right now...

Reply:
I've mixed things up in PR https://github.com/DependencyTrack/dependency-track/pull/2537 with this one... I've pushed the right code in the other PR now